lib/Base/Organization/HomeTimeline.vala

```diff
@@ -23,11 +23,35 @@ using GLib;
 /**
  * The reverse chronological timeline with posts from all followed users.
  */
-public abstract class Backend.HomeTimeline : Backend.Collection {
+public abstract class Backend.HomeTimeline : Collection<Object>, PullableCollection {
+
+  /**
+   * The session handling calls for this collection.
+   */
+  public Session session { get; construct; }
 
   /**
    * The Account which timeline is presented.
    */
   public User account { get; construct; }
 
+  /**
+   * The id of the newest item in the collection.
+   */
+  protected string? newest_item_id { get; set; default = null; }
+
+  /**
+   * Run at construction of the class.
+   */
+  class construct {
+    sorter = new ReversePostSorter ();
+  }
+
+  /**
+   * Calls the API to retrieve all posts from this timeline.
+   *
+   * @throws Error Any error while accessing the API and pulling the posts.
+   */
+  public abstract async void pull_items () throws Error;
```

Comment:
Is this not defined on line 38 of `CollectionExtensions.vala`? Or am I missing something about how Vala inheritance works?

Or is it purely about the doc string and being more specific about it being posts?

Reply:
Vala requires any inheritor of an interface to override its abstract elements. Even if you want to keep it abstract.

I guess this is needed because of Vala compiling that down to C methods.

Reply:
Fair enough 😁 Sorry - I've been programming for so long that I can pick up most languages, but it also means that I know so many bits of so many that I don't always know the odd patterns that they enforce and just find it through compiler errors!